quic/s2n-quic-core/src/recovery/bbr/probe_rtt.rs

```diff
@@ -90,56 +62,87 @@ impl BbrCongestionController {
         //#         BBRCheckProbeRTTDone()
 
         // Ignore low rate samples during ProbeRTT
-        self.bw_estimator.on_app_limited(bytes_in_flight);
+        bw_estimator.on_app_limited(bytes_in_flight);
 
-        if !self.probe_rtt_state.timer.is_armed() && bytes_in_flight <= self.probe_rtt_cwnd() {
+        if !self.timer.is_armed() && bytes_in_flight <= probe_rtt_cwnd {
             // Wait for at least ProbeRTTDuration to elapse:
-            self.probe_rtt_state.timer.set(now + PROBE_RTT_DURATION);
+            self.timer.set(now + PROBE_RTT_DURATION);
             // Wait for at least one round to elapse:
-            self.probe_rtt_state.round_done = false;
-            self.round_counter
-                .set_round_end(self.bw_estimator.delivered_bytes());
-        } else if self.probe_rtt_state.timer.is_armed() {
-            if self.round_counter.round_start() {
-                self.probe_rtt_state.round_done = true;
-            }
-            if self.probe_rtt_state.round_done {
-                self.check_probe_rtt_done(random_generator, now)
-            }
+            self.round_done = false;
+            round_counter.set_round_end(bw_estimator.delivered_bytes());
+        } else if self.timer.is_armed() && round_counter.round_start() {
+            self.round_done = true;
         }
     }
 
-    /// Checks if the ProbeRtt state should be exited and calls `exit_probe_rtt` if so
-    ///
-    /// The next ProbeRtt will be scheduled as necessary
-    pub fn check_probe_rtt_done<Rnd: random::Generator>(
+    /// Returns true if the `ProbeRtt` state is done and should be exited
+    pub fn is_done(&self, now: Timestamp) -> bool {
+        self.round_done && self.timer.is_expired(now)
+    }
+}
+
+/// Methods related to the `ProbeRtt` state
+impl BbrCongestionController {
+    /// Check if it is time to start probing for RTT changes, and enter the ProbeRtt state if so
+    pub fn check_probe_rtt<Rnd: random::Generator>(
```

Comment:
Should all of these methods be public? Or should they only be referenced through the congestion::Controller trait?

Reply:
would `pub(super)` be appropriate here? Making them private doesn't work since I've split the `BbrCongestionController` impl into multiple modules